velox/functions/lib/SIMDComparisonUtil.h

```diff
@@ -0,0 +1,299 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/expression/VectorFunction.h"
+
+namespace facebook::velox::functions {
+
+namespace detail {
+
+template <typename T, bool kIsConstant>
+inline auto loadSimdData(const T* rawData, vector_size_t offset) {
+  using d_type = xsimd::batch<T>;
+  if constexpr (kIsConstant) {
+    return xsimd::broadcast<T>(rawData[0]);
+  }
+  return d_type::load_unaligned(rawData + offset);
+}
+
+inline uint64_t to64Bits(int8_t* resultData) {
+  using d_type = xsimd::batch<int8_t>;
+  constexpr auto numScalarElements = d_type::size;
+  uint64_t res = 0UL;
+  if constexpr (numScalarElements == 64) {
+    res = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+  } else if constexpr (numScalarElements == 32) {
+    auto* addr = reinterpret_cast<uint32_t*>(&res);
+    *(addr) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+    *(addr + 1) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 32)));
+  } else if constexpr (numScalarElements == 16) {
+    auto* addr = reinterpret_cast<uint16_t*>(&res);
+    *(addr) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData)));
+    *(addr + 1) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 16)));
+    *(addr + 2) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 32)));
+    *(addr + 3) = simd::toBitMask(
+        xsimd::batch_bool<int8_t>(d_type::load_unaligned(resultData + 48)));
+  } else {
+    VELOX_UNSUPPORTED(fmt::format(
```

Comment:
Can we put `static_assert` here to fail it at compile time?

Reply:
Updated, thanks